File: src/main/kotlin/xyz/deseteral/lebkuchenfm/domain/commands/model/Command.kt
package xyz.deseteral.lebkuchenfm.domain.commands.model

import xyz.deseteral.lebkuchenfm.domain.commands.CommandProcessor

data class Command(val key: String, private val rawArgs: String) {
    fun getArgsByDelimiter(delimiter: String): List<String> {
        return rawArgs.split(delimiter).filter { it.isNotEmpty() }
    }
}

fun matchProcessor(command: Command, commandProcessor: CommandProcessor): Boolean {
    return command.key == commandProcessor.key || command.key == commandProcessor.shortKey
}

File: packages/service/src/domain/commands/command.ts
interface Command {
  key: string,
  rawArgs: string,
}

function getCommandArgsByDelimiter(command: Command, delimiter: string): string[] {
  return command.rawArgs
    .split(delimiter)
    .filter((s) => (s.length > 0));
}

export default Command;
export {
  getCommandArgsByDelimiter,
};
